src/runtime_src/core/pcie/linux/system_linux.h

```diff
@@ -46,6 +46,32 @@ class system_linux : public system_pcie
   {
     return monitor_access_type::ioctl;
   }
+
+public:
+  virtual
+  std::shared_ptr<pcidev::pci_device>
+  get_pcidev(unsigned index, bool is_user = true) const;
+
+  virtual
+  size_t
+  get_num_dev_ready(bool is_user) const;
+
+  virtual
+  size_t
+  get_num_dev_total(bool is_user) const;
+
+protected:
+  void
+  register_driver(std::shared_ptr<pcidrv::pci_driver> driver);
+
+private:
+  std::vector<std::shared_ptr<pcidrv::pci_driver>> driver_list;
+
+  std::vector<std::shared_ptr<pcidev::pci_device>> user_ready_list;
+  std::vector<std::shared_ptr<pcidev::pci_device>> user_nonready_list;
+
+  std::vector<std::shared_ptr<pcidev::pci_device>> mgmt_ready_list;
+  std::vector<std::shared_ptr<pcidev::pci_device>> mgmt_nonready_list;
 };
```

Comment:
v

Reply:
After a user get a shared ptr of a dev through system_linux, it's shared ownership between the user and the list. If we ever want to be able to dynamically remove a dev from the list, we can do it by removing the dev from the list without keeping track of the users by ourselves. Of course, we need to add a lock to protect these lists and stop using index as id by then.